### PR TITLE
Add Disease Observer Model 0.1

### PR DESCRIPTION
**Pull request by patricktnast**
### Description

- *Category*:  Observers
- *JIRA issue*: [MIC-4115](https://jira.ihme.washington.edu/browse/MIC-4115)
- *Research reference*:  [Pregnancy Concept Model](https://vivarium-research.readthedocs.io/en/latest/models/concept_models/vivarium_nutrition_optimization/pregnancies/concept_model.html#models)

### Changes and notes
 
Added Disease Observer

### Verification and Testing

Pregnancy is getting observed
CI Passes

### CODE REVIEW

File: src/vivarium_gates_nutrition_optimization/model_specifications/model_spec.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ components:
 #            - DisabilityObserver()
             - MortalityObserver()
             - ResultsStratifier()
+            - DiseaseObserver()
```
Comment (mattkappel) [inline src/vivarium_gates_nutrition_optimization/model_specifications/model_spec.yaml:10]: I would have expected you need the name of the disease component as an arg for `DiseaseObserver`, e.g., `DiseaseObserver("ischemic_stroke")` or `DiseaseObserver("pregnancy")` for this usage.

Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/model_specifications/model_spec.yaml:10]: arg, you are right

Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/model_specifications/model_spec.yaml:10]: I did that in implementation as I was testing and forgot to add it back here

Comment (patricktnast) [reply src/vivarium_gates_nutrition_optimization/model_specifications/model_spec.yaml]: fixed

**Review by mattkappel — APPROVED** [review]

**Pull request merged**